### ISSUE TITLE
Upgrading from 3.10.1 to 3.11 results in errors

### ISSUE THREAD

**Issue by jcsrb** [elastic/apm-agent-ruby#884]
Running Kubernetes in Google Cloud I upgraded from 3.10.1 to 3.11 and started receiving `failed to validate metadata` errors 

```
[ElasticAPM] APM Server responded with an error:
"{\"accepted\":0,\"errors\":[{\"message\":\"failed to validate metadata: error validating JSON: I[#] S[#] doesn't validate with \\\"metadata#\\\"\\n  I[#/cloud/account/id] S[#/properties/cloud/properties/account/properties/id/type] expected string, but got null\",\"document\":\"{\\\"metadata\\\":{\\\"service\\\":{\\\"name\\\":\\\"vlapi\\\",\\\"environment\\\":\\\"staging\\\",\\\"version\\\":\\\"cd5694a1235a8561f64868fa28c2e05080a3d12b\\\",\\\"agent\\\":{\\\"name\\\":\\\"ruby\\\",\\\"version\\\":\\\"3.11.0\\\"},\\\"framework\\\":{\\\"name\\\":\\\"Ruby on Rails\\\",\\\"version\\\":\\\"5.2.4.4\\\"},\\\"language\\\":{\\\"name\\\":\\\"ruby\\\",\\\"version\\\":\\\"2.6.6\\\"},\\\"runtime\\\":{\\\"name\\\":\\\"ruby\\\",\\\"version\\\":\\\"2.6.6\\\"}},\\\"process\\\":{\\\"pid\\\":1,\\\"title\\\":\\\"/app/vendor/bundle/ruby/2.6.0/bin/sidekiq\\\",\\\"argv\\\":[]},\\\"system\\\":{\\\"hostname\\\":\\\"vlapi-stag-0\\\",\\\"architecture\\\":\\\"x86_64\\\",\\\"platform\\\":\\\"linux\\\",\\\"kubernetes\\\":{\\\"namespace\\\":null,\\\"node\\\":{\\\"name\\\":null},\\\"pod\\\":{\\\"name\\\":null,\\\"uid\\\":\\\"27649226-1ce3-480a-8a96-06386ad103ef\\\"}}},\\\"labels\\\":null,\\\"cloud\\\":{\\\"provider\\\":\\\"gcp\\\",\\\"account\\\":{\\\"id\\\":null,\\\"name\\\":null},\\\"availability_zone\\\":\\\"europe-west1-b\\\",\\\"instance\\\":{\\\"id\\\":\\\"3835451427227193579\\\",\\\"name\\\":\\\"gke-vlgke-a-default-pool-00000000-0000\\\"},\\\"machine\\\":{\\\"type\\\":\\\"projects/500000000000/machineTypes/n1-standard-4\\\"},\\\"project\\\":{\\\"id\\\":\\\"500000000000\\\",\\\"name\\\":\\\"my-project-512\\\"},\\\"region\\\":\\\"europe-west1\\\"}}}\"}]}\n"
```

## Environment

- **OS**: 
- **Ruby version**: Ruby 2.6.6
- **Framework and version**: Rails 5.2.4.4)
 **APM Server version**: 7.9.2
- **Agent version**: 3.11

maybe related to #882

**Comment by mikker** [comment]
Thank you for reporting this.

This is a bug in APM Server (https://github.com/elastic/apm-server/pull/4142). It is fixed in 7.10.0 which will be released soon.

I'll make a quickfix release of the agent to fix compatibility until then.

**Comment by mikker** [comment]
Let me know if `v3.11.1` fixes this for you and thanks again for reporting it.
